src/coreComponents/mesh/generators/CellBlockManagerABC.hpp

```diff
@@ -0,0 +1,187 @@
+   */
+  static constexpr int maxEdgesPerNode()
+  { return 200; }
+
+  /**
+   * @brief Extra space for node to edges mapping.
+   * @return Number of extra values as an integer.
+   */
+  static constexpr localIndex getEdgeMapOverallocation()
+  { return 8; }
+
+  /**
+   * @brief Extra space for node to elements mapping.
+   * @return Number of extra values as an integer.
+   */
+  static constexpr localIndex getElemMapOverAllocation()
+  { return 8; }
+
+  /**
+   * @brief Extra space for extra nodes.
+   * @return Number of extra values as an integer.
+   */
+  static constexpr localIndex nodeMapExtraSpacePerFace()
+  { return 4; }
+
+  /**
+   * @brief Extra space for extra faces.
+   * @return Number of extra values as an integer.
+   */
+  static constexpr localIndex edgeMapExtraSpacePerFace()
+  { return 4; }
+
+  /**
+   * @brief Extra space for extra edges.
+   * @return Number of extra values as an integer.
+   */
+  static constexpr localIndex faceMapExtraSpacePerEdge()
+  { return 4; }
+
+  /**
+   * @brief Returns a group containing the cell blocks as CellBlockABC instances
+   * @return Mutable reference to the cell blocks group.
+   *
+   * @note It should probably be better not to expose a non-const accessor here.
+   */
+  virtual Group & getCellBlocks() = 0;
+
+  /**
+   * @brief Total number of nodes across all the cell blocks.
+   * @return The total number of nodes.
+   *
+   * Nodes shared by multiple cell blocks are counted only once.
+   */
+  virtual localIndex numNodes() const = 0;
+
+  /**
+   * @brief Total number of edges across all the cell blocks.
+   * @return The total number of edges.
+   */
+  virtual localIndex numEdges() const = 0; // TODO Improve doc
+
+  /**
+   * @brief Total number of faces across all the cell blocks.
+   * @return The total number of faces.
+   */
+  virtual localIndex numFaces() const = 0; // TODO Improve doc
+
+  /**
+   * @brief Returns the node coordinates in a (numNodes, 3) 2d array.
+   * @return A const view to the array.
+   */
+  virtual array2d< real64, nodes::REFERENCE_POSITION_PERM > const & getNodesPositions() const = 0;
+
+  /**
+   * @brief Returns the node to edges mapping.
+   * @return The one to many relationship.
+   */
+  virtual ArrayOfSets< localIndex > getNodeToEdges() const = 0;
+
+  /**
+   * @brief Returns the face to nodes mappings.
+   * @return The one to many relationship.
+   */
+  virtual ArrayOfSets< localIndex > getNodeToFaces() const = 0;
+
+  /**
+   * @brief Returns the node to elements mapping.
+   * @return A one to many relationship.
+   *
+   * @note The mapping is computed on the fly and returned. It is not stored in the instance.
+   */
+  virtual ArrayOfArrays< localIndex > getNodeToElements() const = 0;
+
+  /**
+   * @brief Returns the edge to nodes mapping.
+   * @return A 1 to 2 relationship. The result is meant to have size (numEdges, 2).
+   */
+  virtual array2d< geosx::localIndex > const & getEdgeToNodes() const = 0;
```

Comment:
We already talk about that but I found kinda disturbing that we have sometime `array2d` and sometime `ArrayOfArrays`

Reply:
Here, it's kind of simple to have on edge with two nodes. Hence the `array2d` providing a fixed size in both dimensions.

You may be interested by https://github.com/GEOSX/GEOSX/pull/1418#discussion_r631991811